java/native/src/Accessors.cpp

```diff
@@ -0,0 +1,8 @@
+#include "Accessors.hpp"
+
+jfieldID getHandleField(JNIEnv *env, jobject obj)
+{
+    jclass c = env->GetObjectClass(obj);
+    // J is the type signature for long:
+    return env->GetFieldID(c, "nativeHandle", "J");
```

Comment:
An example how we can store pointers, not sure that it's required. However I decided to include it into this PR. Feel free to change everything _completely_